src/discof/repair/fd_repair_tile.c

```diff
@@ -414,6 +550,108 @@ fd_repair_send_requests( fd_repair_tile_ctx_t *   ctx,
   }
 }
 
+static void
+fd_repair_send_request_async( fd_repair_tile_ctx_t *   ctx,
+                              fd_stem_context_t *      stem FD_PARAM_UNUSED,
+                              fd_repair_t *            glob,
+                              fd_repair_out_ctx_t *    sign_out,
+                              enum fd_needed_elem_type type,
+                              ulong                    slot,
+                              uint                     shred_index,
+                              fd_pubkey_t const      * recipient,
+                              long                     now ){
+    fd_active_elem_t * peer = fd_active_table_query(glob->actives, recipient, NULL);
+    if (!peer) return;
+
+    /* Check if the pending sign request map and pool are initialized */
+    if( FD_UNLIKELY( !ctx->pending_sign_req_map || !ctx->pending_sign_req_pool ) ) {
+        FD_LOG_WARNING(( "pending_sign_req_map or pending_sign_req_pool is NULL" ));
+        return;
+    }
```

Comment:
LOG_ERR is more appropriate here, but it's not really a necessary check. initialization should be done in unprivileged_init and it should have failed there